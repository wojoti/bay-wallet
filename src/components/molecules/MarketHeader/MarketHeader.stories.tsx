import {ComponentMeta, ComponentStory} from '@storybook/react-native';
import MarketHeader from './MarketHeader';

export default {
  title: 'Molecules/MarketHeader',
  component: MarketHeader,
  argTypes: {onButtonPress: {action: 'onButtonPress'}},
} as ComponentMeta<typeof MarketHeader>;
const Template: ComponentStory<typeof MarketHeader> = args => (
  <MarketHeader {...args} />
);

export const Example = Template.bind({});
Example.args = {header: 'Latest Market', buttonLabel: 'View more', testId: ''};
