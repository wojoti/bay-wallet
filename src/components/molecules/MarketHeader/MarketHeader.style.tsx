import {StyleSheet} from 'react-native';
import {primary} from 'theme/colors';

const style = StyleSheet.create({
  wrapper: {marginTop: 20},
  btnStyle: {
    borderWidth: 0,
    height: 50,
  },
  btnTextStyle: {
    fontSize: 14,
    color: primary,
  },
});

export default style;
