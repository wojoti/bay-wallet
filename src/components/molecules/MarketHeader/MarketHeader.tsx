import {Button, Container, Header} from 'components/atoms';
import {PropsWithChildren} from 'react';
import style from './MarketHeader.style';

export type MarketHeaderProps = PropsWithChildren<{
  header: string;
  buttonLabel: string;
  onButtonPress: () => void;
  testId?: string;
}>;

const MarketHeader = ({
  header,
  buttonLabel,
  onButtonPress,
  testId,
}: MarketHeaderProps) => {
  return (
    <Container
      testId={testId}
      direction="row"
      align="center"
      justify="space-around"
      customStyle={style.wrapper}>
      <Header size={20} testId="marketheader-header">
        {header}
      </Header>
      <Button
        customStyle={style.btnStyle}
        customFontStyle={style.btnTextStyle}
        onPress={onButtonPress}
        testId="marketheader-button">
        {buttonLabel}
      </Button>
    </Container>
  );
};

export default MarketHeader;
