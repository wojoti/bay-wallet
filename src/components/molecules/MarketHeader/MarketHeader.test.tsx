import '@testing-library/jest-native/extend-expect';
import {fireEvent, render, screen, within} from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer';
import MarketHeader, {MarketHeaderProps} from './MarketHeader';

const handlePress = jest.fn();
const props: MarketHeaderProps = {
  onButtonPress: handlePress,
  header: 'header',
  buttonLabel: 'label',
  testId: 'test-marketheader-id',
};

test('should match snapshot', () => {
  const tree = TestRenderer.create(<MarketHeader {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render marketheader', () => {
  render(<MarketHeader {...props} />);
  const marketheaderElement = screen.getByTestId('test-marketheader-id');
  expect(marketheaderElement).toBeOnTheScreen();
});

test('should render marketheader - header', () => {
  render(<MarketHeader {...props} />);
  const marketheaderElement = screen.getByTestId('test-marketheader-id');
  expect(marketheaderElement).toBeOnTheScreen();
  const marketheaderHeaderElement = within(marketheaderElement).getByTestId(
    'marketheader-header',
  );
  expect(marketheaderHeaderElement).toBeOnTheScreen();
  expect(marketheaderHeaderElement).toHaveTextContent('header');
});

test('should render marketheader - button', () => {
  render(<MarketHeader {...props} />);
  const marketheaderElement = screen.getByTestId('test-marketheader-id');
  expect(marketheaderElement).toBeOnTheScreen();
  const marketheaderButtonElement = within(marketheaderElement).getByTestId(
    'marketheader-button',
  );
  expect(marketheaderButtonElement).toBeOnTheScreen();
  const labelElement = within(marketheaderButtonElement).getByTestId(
    'button-text',
  );
  expect(labelElement).toBeOnTheScreen();
  expect(labelElement).toHaveTextContent('label');
});

test('should render marketheader - button - onPress', () => {
  render(<MarketHeader {...props} />);
  const marketheaderElement = screen.getByTestId('test-marketheader-id');
  expect(marketheaderElement).toBeOnTheScreen();
  const marketheaderButtonElement = within(marketheaderElement).getByTestId(
    'marketheader-button',
  );
  expect(marketheaderButtonElement).toBeOnTheScreen();
  expect(handlePress).toBeCalledTimes(0);
  fireEvent.press(marketheaderButtonElement);
  expect(handlePress).toBeCalledTimes(1);
});
