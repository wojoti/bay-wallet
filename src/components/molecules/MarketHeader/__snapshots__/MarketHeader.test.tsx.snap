// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": undefined,
        "flexDirection": "row",
        "justifyContent": "space-around",
        "width": "100%",
      },
      {
        "marginTop": 20,
      },
    ]
  }
  testID="test-marketheader-id"
>
  <Text
    style={
      {
        "color": "black",
        "fontFamily": "Poppins-Bold",
        "fontSize": 20,
        "letterSpacing": undefined,
        "textAlign": "left",
      }
    }
    testID="marketheader-header"
  >
    header
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 8,
          "borderStyle": "solid",
          "borderWidth": 1,
          "height": 65,
          "justifyContent": "center",
          "opacity": 1,
        },
        false,
        {
          "borderWidth": 0,
          "height": 50,
        },
      ]
    }
    testID="marketheader-button"
  >
    <Text
      style={
        [
          {
            "fontFamily": "Poppins-Bold",
            "fontSize": 18,
            "textAlign": "center",
          },
          {
            "color": "#363062",
            "fontSize": 14,
          },
        ]
      }
      testID="button-text"
    >
      label
    </Text>
  </View>
</View>
`;
